### PR TITLE
test(multiple): added e2e tests for filter and search behaviors

### PR DESCRIPTION
**Pull request by sheridansunier**
https://jira.corp.adobe.com/browse/MWPW-137132
https://jira.corp.adobe.com/browse/MWPW-136771
https://jira.corp.adobe.com/browse/MWPW-137140
https://jira.corp.adobe.com/browse/MWPW-137141
https://jira.corp.adobe.com/browse/MWPW-137142

### CODE REVIEW

File: e2e-tests/specs/all.e2e.js [hunk 1]
```diff
@@ -387,17 +380,21 @@ describe('Filter and Search Behaviors', async () => {
...
+        expect(newUrl).toEqual(`${serverPath}/html/e2e/e2e-grid.html?state=${state}&ch_Topic=Stock%252CPersonalization`);
     });
-    it('MWPW-137141: can clear a singluar selected filter from checkbox', async () => {
+    it('MWPW-137141: can clear a singluar selected filter from checkbox and see the updated results', async () => {
         const cloneConfig = structuredClone(config);
         const state = btoa(JSON.stringify(cloneConfig));
         const url = `${serverPath}/html/e2e/e2e-grid.html?state=${state}&sh_=Ze&ch_Topic=Stock%252CPersonalization`;
         await browser.url(url);
-        await browser.setTimeout({ script: 600000000 });
+        await browser.setTimeout({ script: 500000000 });
```
Comment (gmirijan) [inline e2e-tests/specs/all.e2e.js:391]: @sheridansunier I guess this long pause is not intentional.

Comment (sheridansunier) [reply e2e-tests/specs/all.e2e.js:391]: lol no i'll fix that

File: e2e-tests/specs/all.e2e.js [hunk 2]
```diff
@@ -307,34 +346,74 @@ describe('Hide CTA(s):', async () => {
...
+        const state = btoa(JSON.stringify(cloneConfig));
+        const url = `${serverPath}/html/e2e/e2e-grid.html?state=${state}&sh_=Ze&ch_Topic=Stock%252CPersonalization`;
         await browser.url(url);
         await browser.setTimeout({ script: 50000 });
-        const grid = await $('.consonant-CardsGrid');
-        await expect(grid).toHaveChildren(8);
+        const clearAll = await $('.consonant-LeftFilters-clearLink');
+        await clearAll.click();
+        const cardRes = await $('.consonant-LoadMore-text').getText();
+        expect(cardRes).toEqual('Showing 5 of 8 cards');
```
Comment (gmirijan) [inline e2e-tests/specs/all.e2e.js:369]: @sheridansunier Why showing 5 of 8 cards is the correct outcome? if you clear all filters then all 8 should be showing .. unless search filter is not removed.

Comment (sheridansunier) [reply e2e-tests/specs/all.e2e.js:369]: b/c of pagination

File: e2e-tests/specs/all.e2e.js [hunk 3]
```diff
@@ -43,7 +43,7 @@ const config = {
     },
     featuredCards: ['c7d34f39-397c-3727-9dff-5d0d9d8cf731'],
     filterPanel: {
-        enabled: 'false',
+        enabled: 'true',
```
Comment (gmirijan) [inline e2e-tests/specs/all.e2e.js:46]: @sheridansunier  "Carousel" layout will not work if filter panel enable. It is better to enable for filter tests.

image

File: e2e-tests/specs/all.e2e.js [hunk 4]
```diff
@@ -252,41 +252,80 @@ if (args.includes('env=LOCAL')) {
...
+describe('CTA Behaviors', async () => {
+    /* it('CTA should exist', async () => {
         const cloneConfig = structuredClone(config);
         const state = btoa(JSON.stringify(cloneConfig));
-        const url = `${serverPath}/html/e2e/MWPW-126169.html?state=${state}`;
+        const url = `${serverPath}/html/e2e/e2e-grid.html?state=${state}`;
         await browser.url(url);
         await browser.setTimeout({ script: 50000 });
-        const buttonText = await $('#ac578dee-f01b-3ea0-a282-2116619e4251 .consonant-BtnInfobit--cta').getText();
-        expect(buttonText).toEqual('Read More');
```
Comment (gmirijan) [inline e2e-tests/specs/all.e2e.js:263]: Looks like this is removed. Could you please keep it inside the comments? 
expect(buttonText).toEqual('Read More');

File: e2e-tests/specs/all.e2e.js (hunk 3 above)
Comment (sheridansunier) [reply e2e-tests/specs/all.e2e.js:46]: that makes sense - but also the carousel has it's own separate configuration so I feel like it's fine?

**Review by gmirijan — APPROVED** [review]

**Pull request merged**